src/main/java/Problem3/Movie.java

```diff
@@ -9,15 +16,25 @@
 
     public Movie(String rating, String title) {
         this.id = UUID.randomUUID();
-        // homework
+        this.title = title;
+        this.rating = rating;
     }
 
+    // copy constructor
     public Movie(Movie anotherMovie) {
-        // homework
+        this.id = anotherMovie.id;
+        this.title = anotherMovie.title;
+        this.rating = anotherMovie.rating;
     }
 
+    // equals method for comparing variables
     @Override
-    public boolean equals(Object obj) {
-        // homework
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Movie otherMovie = (Movie) o;
+        return rating.equals(otherMovie.rating) &&
+                title.equals(otherMovie.title) &&
+                id.equals(otherMovie.id);
```

Comment:
same issue here. this logic check is wrong

"A movie equals another movie only when their id are the same. Same rule for books being eqaul."

Reply:
different books or movies with the same title and rating will have different ids